app/windows/winget/winget-create.js

```diff
@@ -0,0 +1,149 @@
+ */
+
+const fetch = require('node-fetch');
+const fs = require('fs');
+const path = require('path');
+
+const args = process.argv.slice(2);
+
+if (!(args.length === 2)) {
+  console.error('Usage: node winget-create.js <projectVersion> <directoryPath>');
+  process.exit(1);
+}
+
+const projectVersion = args[0];
+const directoryPath = args[1];
+
+if (projectVersion.startsWith('v')) {
+  console.error('Please provide the project version without the "v" prefix.');
+  console.error('Example: node winget-create.js 0.19.1 my/path/to/folder');
+  process.exit(1);
+}
+
+let sha256;
+
+const arch = {
+  x64: 'x64',
+};
+
+async function createYAMLFiles(version, downloadURL, date, notesURL) {
+  const templates = [
+    'Headlamp.Headlamp.installer.yaml',
+    'Headlamp.Headlamp.locale.en-US.yaml',
+    'Headlamp.Headlamp.yaml',
+  ];
+
+  const templateDir = path.join(__dirname, 'templates/');
+  const manifestDir = path.join(directoryPath, version);
+
+  try {
+    // create manifest version directory if it doesnt exist
+    if (!fs.existsSync(manifestDir)) {
+      fs.mkdirSync(manifestDir, { recursive: true });
+      console.log(`Created directory ${manifestDir}`);
+    }
+  } catch (err) {
+    console.error(`Error creating directory ${manifestDir}: ${err}`);
+    process.exit(1);
+  }
+
+  templates.forEach(template => {
+    const manifestPath = path.join(manifestDir, template);
+    const templatePath = path.join(templateDir, template);
+    const templateContent = fs.readFileSync(templatePath, 'utf8');
+
+    const newContent = templateContent
+      .replace(/__PACKAGE_VERSION__/g, version)
+      .replace('__DISPLAY_VERSION__', version)
+      .replace('__INSTALLER_SHA256__', sha256)
+      .replace('__INSTALLER_URL__', downloadURL)
+      .replace('__RELEASE_DATE__', date)
+      .replace('__RELEASE_NOTES_URL__', notesURL);
+
+    fs.writeFileSync(manifestPath, newContent);
+    console.log(
+      `Created winget manifest file ${template} for version ${version} in ${manifestDir}`
+    );
+  });
+}
+
+// Async function to fetch release information from GitHub API for a given tag
+async function fetchGithubReleaseInfo() {
+  const tag = projectVersion;
+  const res = await fetch(
+    `https://api.github.com/repos/headlamp-k8s/headlamp/releases/tags/v${tag}`
+  );
+
+  if (!res.ok) {
+    console.error(
+      `Error fetching release information for tag ${tag}: ${res.status} ${res.statusText}`
+    );
+    process.exit(1);
+  }
+
+  const resJSON = await res.json();
+
+  await updateReleaseInfo(resJSON, tag);
+}
+
+async function getChecksums(checksumsUrl, fileName) {
+  const res = await fetch(checksumsUrl);
+  const checksums = await res.text();
+  const checksumMap = {};
+
+  for (const line of checksums.split('\n')) {
+    const [checksum, filename] = line.split('  ');
+
+    if (!filename) {
+      continue;
+    }
+
+    if (filename === fileName) {
+      checksumMap[arch.x64] = checksum;
```

Comment:
There's no point in submitting a map when we only have one value.

Reply:
done!